docs/topic/billing/things.md

```diff
@@ -0,0 +1,610 @@
+Depending on the source and destination of this transmission, **we may get charged for it**,
+so it's important to have a good mental model of how network costs work when
+thinking about cloud costs.
+
+While the overall topic of network costs in the cloud is complex, thankfully
+we can get away with a simplified model when operating just JupyterHubs.
+
+### Network Boundaries
+
+There are two primary aspects that determine how much a network
+transmission costs:
+
+1. The total amount of data transmitted
+2. The **network boundaries** it crosses
+
+(1) is fairly intuitive - the more data that is transmitted, the
+more it costs. Usually, this is measured in 'GiB of data transmitted',
+rather than smaller units.
+
+(2) is determined by the **network boundaries** a data packet has to
+cross from source to destination. Network boundaries can be categorized
+in a million ways. But given we primarily want to model network costs,
+we can consider a boundary to be crossed when:
+
+1. The source and destination are in different zones of the same region of the cloud provider
+2. The source and destination are in different regions of the same cloud provider
+3. One of the source or destination is *outside* the cloud provider
+
+If the source and destination are within the same zone, no network cost
+boundaries are crossed and usually this costs no money. As a general rule,
+when each boundary gets crossed, the cost of the data transmission
+goes up. Within a cloud provider, transmission between regions is often
+priced based on the distance between them - australia to
+the US is more expensive than canada to the US. Data transmission from within a cloud provider to outside is the most expensive, and is also often
+metered by geographical location.
+
+This also applies to data present in any *cloud services* as well, although
+*sometimes* in those cases transmission is free even in a different zone in
+the same region. For example, with [AWS S3 data transfer](https://aws.amazon.com/s3/pricing/) (under the 'Data Transfer' tab), accessing
+S3 buckets **in the same region** is free regardless of what zone you are in.
+But, with [GCP Filestore](https://cloud.google.com/filestore/pricing#network_pricing_for_traffic_between_the_client_and), you
+*do* pay network fees if you are mounting it from a different zone even if
+it is in the same region. So as a baseline, we should assume that any
+traffic crossing network *zone* boundaries will be charged, unless explicitly
+told otherwise.
+
+Primarily, this means we should design and place services as close to each
+other as possible to reduce cost.
+
+### Ingress and Egress fees
+
+Cloud providers in general don't want you to leave them, and tend to
+enforce this lock-in by charging an [extremely high network egress cost](https://blog.cloudflare.com/aws-egregious-egress).
+Like, 'bankrupt your customer because they tried to move a bucket of data from
```

Comment:
Should we add something about EU's Digital Markets Act and how cloud may be responding such concerns https://aws.amazon.com/blogs/aws/free-data-transfer-out-to-internet-when-moving-out-of-aws/

Reply:
Done!